Makefile

```diff
@@ -1,23 +1,22 @@
-# fall back to git hash for local builds
-RELEASE_VERSION ?= $(shell git rev-parse --short HEAD)
+APP_NAME	    	:= sam-golang-multi-lambda
 
-LDFLAGS := -ldflags "-s -w -X main.commit=$(VERSION)"
-BUILDFLAGS := -tags lambda.norpc
+GIT_HASH    		?= $(shell git rev-parse --short HEAD)
+STAGE	    		?= dev
+BRANCH      		?= master
+
+LDFLAGS := -ldflags=-X=main.commit=$(GIT_HASH) -ldflags=-s -ldflags=-w
```

Comment:
I _think_ (have to double check) that this won't actually work. I think each `-ldflags=` will overwrite the previous one.

Reply:
Also (just in case you didn't know) instead of using `-X=main.commit=...` you can do the following:

```go
func gitcommit() string {
	bi, ok := debug.ReadBuildInfo()
	if !ok {
		return "unknown"
	}

	for _, bs := range bi.Settings {
		if bs.Key == "vcs.revision" {
			return bs.Value
		}
	}

	return "unknown"
}
```

Reply:
yeah I am just using git commit to demonstrate passing through a version, there are definitely better ways to get this value after Go 1.18.